internal/characters/neuvillette/charge.go

```diff
@@ -0,0 +1,265 @@
+			AnimationLength: windup + chargeFrames[action.InvalidAction],
+			CanQueueAfter:   windup + chargeFrames[action.ActionDash],
+			State:           action.ChargeAttackState,
+		}, nil
+	}
+
+	c.chargeAi = combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       chargeJudgementName,
+		AttackTag:  attacks.AttackTagExtra,
+		ICDTag:     attacks.ICDTagExtraAttack,
+		ICDGroup:   attacks.ICDGroupDefault,
+		StrikeType: attacks.StrikeTypeDefault,
+		Element:    attributes.Hydro,
+		Durability: 25,
+		FlatDmg:    chargeJudgement[c.TalentLvlAttack()] * c.MaxHP(),
+	}
+
+	chargeJudgementStart := windup + chargeLegalEvalLeft
+	chargeJudgementDur := 173
+
+	// if c.Base.Cons >= 6 {
+	// the c6 droplet check has to happen immediately because otherwise we don't know how long this action will take
+	// this is problematic with Q because the 3 of the 6 Q orbs spawn after CA starts.
+	// chargeJudgementDur += c.c6DropletCheck()
+	// }
+	if p["ticks"] > 0 {
+		// param for letting the user not do the full channel
+		// calculate how long the judgement duration should be based on their tick count
+		// additionally modify the frames so that only D/J/Q/E can follow. Otherwise sim errors on action
+		// also need to verify it for c6
+		maxTicks := p["ticks"]
+		ticksDone := 0
+		delay := getChargeJudgementHitmarkDelay(ticksDone)
+		for delay < chargeJudgementDur {
+			c.QueueCharTask(c.judgementWave, chargeJudgementStart+delay)
+			ticksDone += 1
+			delay = getChargeJudgementHitmarkDelay(ticksDone)
+
+			if ticksDone == maxTicks {
+				for d := 40; d < delay; d += 30 {
+					c.QueueCharTask(c.consumeHp, chargeJudgementStart+d)
+				}
+				c.chargeEarlyCancelled = true
+				return action.Info{
+					Frames: func(next action.Action) int {
+						return chargeJudgementStart + delay + earlyCancelEndLag[next]
+					},
+					AnimationLength: chargeJudgementStart + delay + earlyCancelEndLag[action.InvalidAction],
+					CanQueueAfter:   chargeJudgementStart + delay + earlyCancelEndLag[action.ActionDash],
+					State:           action.ChargeAttackState,
+				}, nil
+			}
+		}
+		if maxTicks == ticksDone+1 {
+			c.QueueCharTask(c.judgementWave, chargeJudgementStart+chargeJudgementDur)
+			// He drains 5 times in 3s, on frame 40, 70, 100, 130, 160
+			for d := 40; d < chargeJudgementDur; d += 30 {
+				c.QueueCharTask(c.consumeHp, chargeJudgementStart+d)
+			}
+			return action.Info{
+				Frames: func(next action.Action) int {
+					return chargeJudgementStart + chargeJudgementDur + endLag[next]
+				},
+				AnimationLength: chargeJudgementStart + chargeJudgementDur + endLag[action.InvalidAction],
+				CanQueueAfter:   chargeJudgementStart + chargeJudgementDur + endLag[action.ActionDash],
+				State:           action.ChargeAttackState,
+			}, nil
+		}
+
+		return action.Info{}, fmt.Errorf("%v: Cannot execute %d CA Judgement Ticks. Max executed %d", c.CharWrapper.Base.Key, maxTicks, ticksDone)
+	}
```

Comment:
nit:
```go
	if p["ticks"] > 0 {
		// param for letting the user not do the full channel
		// calculate how long the judgement duration should be based on their tick count
		// additionally modify the frames so that only D/J/Q/E can follow. Otherwise sim errors on action
		// also need to verify it for c6
		maxTicks := p["ticks"]
		ticksDone := 0
		delay := getChargeJudgementHitmarkDelay(ticksDone)
		lag := endLag
		for delay < chargeJudgementDur {
			c.QueueCharTask(c.judgementWave, chargeJudgementStart+delay)
			ticksDone++
			delay = getChargeJudgementHitmarkDelay(ticksDone)

			if ticksDone == maxTicks {
				c.chargeEarlyCancelled = true
				lag = earlyCancelEndLag
				break
			}
		}
		if delay > chargeJudgementDur {
			delay = chargeJudgementDur
		}

		
		if maxTicks == ticksDone+1 {
			c.QueueCharTask(c.judgementWave, chargeJudgementStart+chargeJudgementDur)
		} else if maxTicks > ticksDone {
			return action.Info{}, fmt.Errorf("%v: Cannot execute %d CA Judgement Ticks. Max executed %d", c.CharWrapper.Base.Key, maxTicks, ticksDone)
		}

		// He drains 5 times in 3s, on frame 40, 70, 100, 130, 160
		for d := 40; d < delay; d += 30 {
			c.QueueCharTask(c.consumeHp, chargeJudgementStart+d)
		}
		return action.Info{
			Frames: func(next action.Action) int {
				return chargeJudgementStart + delay + lag[next]
			},
			AnimationLength: chargeJudgementStart + delay + lag[action.InvalidAction],
			CanQueueAfter:   chargeJudgementStart + delay + lag[action.ActionDash],
			State:           action.ChargeAttackState,
		}, nil
	}
```